### PR TITLE
Add inclusion/exclusion parameters to completion

### PR DESCRIPTION
**Pull request by volkerstampa**
Completion requests support 4 new parameters regarding inclusion and excludion of tokens to be generated.

### CODE REVIEW

File: aleph_alpha_client/aleph_alpha_client.py [hunk 1]
```diff
@@ -377,6 +389,20 @@ def complete(
...
+            completion_bias_inclusion (List[str], default [])
+                Bias the completion to only generate options within this list;
+                all other tokens are disregarded at sampling
+
+            completion_bias_inclusion_first_token_only (bool, default False)
+                Only consider the first token for the completion_bias_inclusion
+
+            completion_bias_exclusion (List[str], default [])
+                Bias the completion to NOT generate options within this list;
+                all other tokens are disregarded at sampling
```
Comment (ahartel) [inline aleph_alpha_client/aleph_alpha_client.py:402]: all other tokens are allowed at sampling?

File: aleph_alpha_client/completion.py [hunk 2]
```diff
@@ -126,6 +126,26 @@ class CompletionRequest(NamedTuple):
...
+
+            Note that strings in the inclusion list must not be prefixes
+            of strings in the exclusion list and vice versa
+
+        completion_bias_inclusion_first_token_only (bool, default False)
+            Only consider the first token for the completion_bias_inclusion
+
+        completion_bias_exclusion (List[str], default [])
+            Bias the completion to NOT generate options within this list;
+            all other tokens are disregarded at sampling
```
Comment (ahartel) [inline aleph_alpha_client/completion.py:141]: same here

File: aleph_alpha_client/completion.py [hunk 3]
```diff
@@ -156,9 +176,13 @@ class CompletionRequest(NamedTuple):
     sequence_penalty: float = 0.0
     sequence_penalty_min_length: int = 2
     use_multiplicative_sequence_penalty: bool = False
+    completion_bias_inclusion: Optional[Sequence[str]] = None
+    completion_bias_inclusion_first_token_only: bool = False
+    completion_bias_exclusion: Optional[Sequence[str]] = None
+    completion_bias_exclusion_first_token_only: bool = False
 
     def to_json(self) -> Dict[str, Any]:
-        payload = self._asdict()
```
Comment (ahartel) [inline aleph_alpha_client/completion.py:161]: was this change necessary at all? Or is it more like an optimization?

Comment (volkerstampa) [reply aleph_alpha_client/completion.py:161]: On the python side an Option[Sequence] make sense as you typically do not want to have a mutable default value (i.e. []), however `None` gets serialized to a `null` json. On the API side there is no reason to support a _nullable_ field as a non-existing field can simply be deserialized into the default empty list. This change makes sure that we do not send `null`s in json structs for _undefined_ fields. Makes sense?

Comment (ahartel) [reply aleph_alpha_client/completion.py:161]: 👍
